app/scripts/controllers/transactions/index.js

```diff
@@ -193,12 +193,12 @@ class TransactionController extends EventEmitter {
     }
     txUtils.validateTxParams(normalizedTxParams)
     // construct txMeta
-    const { transactionCategory, getCodeResponse } = await this._determineTransactionCategory(txParams)
     let txMeta = this.txStateManager.generateTxMeta({
       txParams: normalizedTxParams,
       type: TRANSACTION_TYPE_STANDARD,
-      transactionCategory,
     })
+    const { transactionCategory, getCodeResponse } = await this._determineTransactionCategory(txParams)
```

Comment:
Interesting - was this change incidental, or is there a subtle change in behaviour here?

Reply:
In that the line has been reordered? I too am curious about this.

Reply:
This change causes the `id` of the transaction to be set before the async call to `this._determineTransactionCategory(txParams)`. This `id` is added with the call to `generateTxMeta`

These `id`s are generated with an increment, so that each successive call to `generateTxMeta` will assign an id that is 1 greater than the previous call.

If we make the async call to `this._determineTransactionCategory` before the id is assigned, the following can occur:
- `addUnapprovedTransaction` called multiple times in short succession
- user or developer expects transactions to be ordered as they are received by `addUnapprovedTransaction`
- the wait time for `await this._determineTransactionCategory` differes for each of these different transactions
- the ids get assigned to the transactions in order of the completion of the async call, not the order with which they were received by `addUnapprovedTransaction`
- the transactions have an unexpected order in the tx-state-manager

Reply:
Oof. Didn't realize that `_determineTransactionCategory` was async. That makes sense.